hey-discount/src/main/java/Ad.java

```diff
@@ -0,0 +1,177 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+class Review {
+    private String username;
+    private Date date;
+    private String reviewText;
+
+    public Review() {
+        username = "";
+        date = new Date();
+        reviewText = "";
+    }
+
+    public Review(String username, Date date, String reviewText) {
+        this.username = username;
+        this.date = date;
+        this.reviewText = reviewText;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setDate(Date date) {
+        this.date = date;
+    }
+
+    public void setReviewText(String reviewText) {
+        this.reviewText = reviewText;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public Date getDate() {
+        return date;
+    }
+
+    public String getReviewText() {
+        return reviewText;
+    }
+}
+
+public class Ad {
+    private String id;
+    private String description;
+    private Date createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private List<String> hashTags;
+    private String discount;
+    private Date validUntil;
+    private Integer rating;
+    private List<Review> reviews;
+
+    public Ad() {
+        id = "";
+        description = "";
+        createdAt = new Date();
+        link = "";
+        vendor = "";
+        photoLink = "";
+        hashTags = new ArrayList<>();
+        discount = "";
+        validUntil = new Date();
+        rating = 0;
+        reviews = new ArrayList<>();
+    }
+
+    public Ad(String id, String description, Date createdAt, String link, String vendor, String photoLink, List<String> hashTags,
```

Comment:
Please be informed that the normal number of arguments per method is 0, or 1, or 2, or 3. 3 is some extreme. If you  have more, you are in some trouble) Eleven args in no way to go!  When in a commercial project at least.
When do people do with such cases?
Create builders. https://habr.com/ru/post/244521/
Or use setters
Or use some means that hide argument initialization away from reader. Namely, Lombok https://habr.com/ru/post/345520/

I'd like you to master the builder pattern, it will help you greatly in the course of time